### ISSUE TITLE
add_reference doesn't find foreign_key to drop when rolling back

### ISSUE THREAD

**Issue by dirceuvjr** [rails/rails#25755]
### Steps to reproduce

Assuming you have a :restaurants table and :users table, try to create a migration to add a reference of user as 'admin' to restaurants:

```ruby
# db/migrate/20160708201927_add_admin_user_to_restaurant.rb
class AddAdminUserToRestaurant < ActiveRecord::Migration[5.0]
  def change
    add_reference :restaurants, :admin, :index => true, :foreign_key => { :to_table => :users }
  end
end
```

### Expected behavior

When running **rails db:rollback STEP=1**, it should find the foreign_key 'admin_id' and drop it.

### Actual behavior

**rails db:rollback STEP=1** failed with the following message **Table 'restaurants' has no foreign key for admins**

### System configuration
**Rails version**: 5.0.0
**Ruby version**: 2.3.1
**Postgres**: 9.5.3

**Comment by rafaelfranca** [comment]
there is no option `to_table` for the `foreign_key` option, where did you saw it in the documentation? Rails will use the second argument of `add_reference` which in your case it is `:admin`. I believe this is why it was not created and I believe it is why it is failing in the rollback with that message.

**Comment by dirceuvjr** [comment]
@rafaelfranca:
http://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-add_reference

This actually works, that's why I changed the issue. It creates the FK, however it fails to drop it when running rollback.

**Comment by rafaelfranca** [comment]
Oh. I was looking the 4.2 documentation. :+1:

**Comment by mechanicles** [comment]
I was able to reproduce this issue on my machine.

```
== 20160712102939 AddUserReferenceToRestaurant: reverting =====================
-- remove_reference(:restaurants, :admin, {:index=>true, :foreign_key=>{:to_table=>:users}})
rails aborted!
StandardError: An error has occurred, this and all later migrations canceled:

Table 'restaurants' has no foreign key for admins
/Users/santosh/my-rails-test-apps/test1/bin/rails:9:in `require'
/Users/santosh/my-rails-test-apps/test1/bin/rails:9:in `<top (required)>'
/Users/santosh/my-rails-test-apps/test1/bin/spring:13:in `require'
/Users/santosh/my-rails-test-apps/test1/bin/spring:13:in `<top (required)>'
bin/rails:3:in `load'
bin/rails:3:in `<main>'
ArgumentError: Table 'restaurants' has no foreign key for admins
/Users/santosh/my-rails-test-apps/test1/bin/rails:9:in `require'
/Users/santosh/my-rails-test-apps/test1/bin/rails:9:in `<top (required)>'
/Users/santosh/my-rails-test-apps/test1/bin/spring:13:in `require'
/Users/santosh/my-rails-test-apps/test1/bin/spring:13:in `<top (required)>'
bin/rails:3:in `load'
bin/rails:3:in `<main>'
Tasks: TOP => db:rollback
(See full trace by running task with --trace)
```

**Comment by mechanicles** [comment]
@rafaelfranca This issue is already solved here https://github.com/rails/rails/issues/25169 and https://github.com/rails/rails/pull/25786

**Comment by prathamesh-sonpatki** [comment]
Fixed by #25931
